octoprint_mrbeam/migrate.py

```diff
@@ -214,6 +215,19 @@ def run(self):
                 ):
                     self.hostname_helper_scripts()
 
+                beamos_tier, beamos_date = self.plugin._device_info.get_beamos_version()
+                if beamos_date <= datetime.strptime(
+                    "2021-06-25", "%Y-%m-%d"
+                ).date() and (
+                    self.version_previous is None
```

Comment:
Also this might raise a None-comparison error if beamos_date is None.

Always assume that some info read from a "config" file can be None

Reply:
the get_beamos_version returns the legacy date if it can't get the configdata, but i added a condition to handle it